src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -340,3 +339,135 @@ def load_secure_props(self) -> None:
+        except keyring.errors.PasswordDeleteError:
+            # keyring.delete_password is not available (e.g., macOS)
+            if sys.platform == "win32":
+                # Delete the secure value with an index
+                index = 1
+                while True:
+                    field_name = f"{account_name}-{index}"
+                    try:
+                        keyring.delete_password(f"{service_name}-{index}", field_name)
+                    except keyring.errors.PasswordDeleteError:
+                        break
+                    index += 1
+    
+    def set_secure_props(self) -> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+        
+        try:
+            # Filter or suppress specific warning messages
+            warnings.filterwarnings("ignore", message="^Retrieved an UTF-8 encoded credential")
+            service_name = constants["ZoweServiceName"]
+            credential = {self.filepath: self.secure_props}
+            # Check if credential is a non-empty string
+            if credential:
+                is_win32 = sys.platform == "win32"
+                if is_win32:
+                    service_name += "/" + constants["ZoweAccountName"] 
+                
+               
+                # Load existing credentials, if any
+                existing_credential = self._retrieve_credential(service_name)
+                if existing_credential:
+                        
+                    # Decode the existing credential and update secure_props
+                    existing_credential_bytes = base64.b64decode(existing_credential).decode()
+                    existing_secure_props = commentjson.loads(existing_credential_bytes)
+                    existing_secure_props[self.filepath].update(credential[self.filepath])
+                    # Encode the credential
+                    encoded_credential = base64.b64encode(commentjson.dumps(existing_secure_props).encode()).decode()
+                    # Delete the existing credential
+                    self.delete_credential(service_name , constants["ZoweAccountName"])
+                else:
+                    print("here")
+                    # Encode the credential
+                    encoded_credential = base64.b64encode(commentjson.dumps(credential).encode()).decode() 
+                print(encoded_credential)  
+                # Check if the encoded credential exceeds the maximum length for win32
+                if is_win32 and len(encoded_credential) > constants["WIN32_CRED_MAX_STRING_LENGTH"]:
+                    # Split the encoded credential string into chunks of maximum length
+                    chunk_size = constants["WIN32_CRED_MAX_STRING_LENGTH"]
+                    chunks = [encoded_credential[i: i + chunk_size] for i in range(0, len(encoded_credential), chunk_size)]
```

Comment:
If the credential exceeds the maximum length on Windows, then we need to add a `\0` byte at the end to indicate end of data before we split it into chunks.

This is separate from the second `\0` that is already present inside the for loop to ensure the string is an even number of characters.